Flaky tests

Sometimes (in 5-10% of GitHub Actions runs), unit and nightly tests fail. I found two flaky tests:

- Nightly tests: [link](https://github.com/sio2project/oioioi/actions/runs/12133571941) - FAILED oioioi/problems/tests/test_problemset.py::TestProblemsetFilters::test_filters
- Unit tests: [link](https://github.com/sio2project/oioioi/actions/runs/12123980565) - FAILED oioioi/sinolpack/tests.py::TestSinolPackage::test_single_file_replacement

The flaky test
> Unit tests: [link](https://github.com/sio2project/oioioi/actions/runs/12123980565) - FAILED oioioi/sinolpack/tests.py::TestSinolPackage::test_single_file_replacement 

has been fixed by PR #442.

--- a/oioioi/problems/tests/test_problemset.py
+++ b/oioioi/problems/tests/test_problemset.py
@@ -219,7 +219,9 @@ class TestProblemsetFilters(TestCase):
             self.assertEqual(response.status_code, 200)
 
             for problem in self.problems:
+                problemTag = f"<td>{problem}</td>"
+
                 if problem in filtered:
-                    self.assertContains(response, problem)
+                    self.assertContains(response, problemTag, html=True)
                 else:
-                    self.assertNotContains(response, problem)
+                    self.assertNotContains(response, problemTag, html=True)

